src/errors.ts

```diff
@@ -1,30 +1,24 @@
 
-export class ConnectionError implements Error {
-    public name = "ConnectionError";
-    public message = "";
+class BaseError {
```

Comment:
Why doesn't `BaseError` implement `Error` to save some trouble? Is there a possibility that having an unimplemented version may be desirable in some circumstances?